### PR TITLE
unit test for BRE interfaces/model

### PR DESCRIPTION
**Pull request by jimlebeau**
## Description

Add unit test for classes in cares-bre-interface 

```
gov.ca.cwds.bre.interfaces.model 
gov.ca.cwds.bre.interfaces.exception
```

Added a ReporterDataBuilder class for developing test of the Reporter business rules in the future.

Updated a couple of Reporter business rules drl files.
Corrected some syntax in Reporter business rules drl files.

## Jira Issue link

https://osi-cwds.atlassian.net/browse/REP-90

### CODE REVIEW

File: cares-bre/cares-bre-interfaces/src/test/java/gov/ca/cwds/bre/interfaces/model/BreResponseDataTest.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+public class BreResponseDataTest {
+
+  @Test
+  public void shouldCreateObjectUsingDefaultConstructor() throws Exception {
+    BreResponseData rep = new BreResponseData();
+    assertThat(rep, notNullValue());
+  }
```
[the remark this answers is not in the document]
Comment (OlegKorniichuk) [reply cares-bre/cares-bre-interfaces/src/test/java/gov/ca/cwds/bre/interfaces/model/BreResponseDataTest.java]: Some thoughts:
1. should we test that new operator returns not null value?
2. should we test POJO beans?

File: cares-bre/cares-bre-interfaces/src/test/java/gov/ca/cwds/bre/interfaces/model/BreRequestDataTest.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  @Test
+  public void shouldSetGetDataObject() throws Exception {
+    String json = "{ \"f1\" : \"v1\" } ";
+    ObjectMapper objectMapper = new ObjectMapper();
+    JsonNode jsonNode = objectMapper.readTree(json);
+    BreRequestData req = new BreRequestData();
+    req.setDataObject(jsonNode);
+    assertEquals(req.getDataObject(), jsonNode);
+  }
+}
```
[the remark this answers is not in the document]
Comment (jimlebeau) [reply cares-bre/cares-bre-interfaces/src/test/java/gov/ca/cwds/bre/interfaces/model/BreRequestDataTest.java]: probably not - but just to verify that correct property is set/get

File: cares-bre/cares-bre-interfaces/src/test/java/gov/ca/cwds/bre/interfaces/exception/BreExceptionTest.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+  public void shouldGetSetBreResponse() throws Exception {
+    BreResponse rep = new BreResponse();
+    String ruleSetName = "rule set name";
+    rep.setBusinessRuleSetName(ruleSetName);
+
+    BreException exception = new BreException();
+    exception.setBreResponse(rep);
+    assertEquals(exception.getBreResponse(), rep);
+  }
+
```
[the remark this answers is not in the document]
Comment (jimlebeau) [reply cares-bre/cares-bre-interfaces/src/test/java/gov/ca/cwds/bre/interfaces/exception/BreExceptionTest.java]: No logic yet.  Verifying that the correct property is set.

File: cares-bre/cares-bre-interfaces/src/test/java/gov/ca/cwds/bre/interfaces/model/BreResponseDataTest.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (jimlebeau) [reply cares-bre/cares-bre-interfaces/src/test/java/gov/ca/cwds/bre/interfaces/model/BreResponseDataTest.java]: Just verifying that constructor is present and works.

File: cares-bre/cares-bre-interfaces/src/test/java/gov/ca/cwds/bre/interfaces/exception/BreExceptionTest.java [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+public class BreExceptionTest {
+
+  String message = "exception message";
+  Throwable cause = new Throwable();
+
+  @Test
+  public void shouldConstructObjecWithDefault() throws Exception {
+    BreException exception = new BreException();
+    assertThat(exception, notNullValue());
+  }
```
[the remark this answers is not in the document]
Comment (jimlebeau) [reply cares-bre/cares-bre-interfaces/src/test/java/gov/ca/cwds/bre/interfaces/exception/BreExceptionTest.java]: Just verifying that constructor is present and works.

File: cares-bre/cares-bre-interfaces/src/test/java/gov/ca/cwds/bre/interfaces/model/BreRequestDataTest.java [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+public class BreRequestDataTest {
+
+  @Test
+  public void shouldConstructUsingDefaultConstructor() throws Exception {
+    BreRequestData req = new BreRequestData();
+    assertThat(req, notNullValue());
+  }
```
[the remark this answers is not in the document]
Comment (jimlebeau) [reply cares-bre/cares-bre-interfaces/src/test/java/gov/ca/cwds/bre/interfaces/model/BreRequestDataTest.java]: Just verifying that constructor is present and works.

File: cares-bre/cares-bre-interfaces/src/test/java/gov/ca/cwds/bre/interfaces/model/BreRequestTest.java [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+public class BreRequestTest {
+
+  @Test
+  public void shouldCreateObjectUsingDefaultConstructor() throws Exception {
+    BreRequest def = new BreRequest();
+    assertThat("BreRequest is not null", def, notNullValue());
+  }
```
[the remark this answers is not in the document]
Comment (jimlebeau) [reply cares-bre/cares-bre-interfaces/src/test/java/gov/ca/cwds/bre/interfaces/model/BreRequestTest.java]: Verifying that constructor is present and works.

**Pull request merged**